lib/client.js

```diff
@@ -62,21 +51,18 @@ class Parser extends HTTPParser {
     const request = client[kQueue][client[kComplete]]
     const skipBody = request.method === 'HEAD'
 
-    const body = new Readable({
-      autoDestroy: true,
-      read: this.bodyRead,
-      destroy: this.bodyDestroy
-    })
-    body.push = request.wrapSimple(body, body.push)
-
-    this.body = body
     this.read = 0
-
-    request.callback(null, {
+    this.body = request.callback(null, {
       statusCode,
-      headers: parseHeaders(headers),
-      body
+      headers: parseHeaders(headers)
     })
+    // TODO: Should only resume if body.write paused.
+    this.body
+      .on('drain', this.resume)
+      .on('error', this.resume)
+      .on('close', this.resume)
+    this.body.write = request.wrapSimple(this.body, this.body.write)
+    this.body.end = request.wrapSimple(this.body, this.body.end)
```

Comment:
I'm not exactly sure of the purpose of wrapSimple... should `destroy` also be wrapped?

Reply:
I'm really -1 on this wrapping. Why is this needed? This looks less maintainable.

Reply:
You mean the `wrapSimple`? I don't know. Just following the old code:

```js
body.push = request.wrapSimple(body, body.push)
```

I'm unsure what exactly this does.

Reply:
https://github.com/mcollina/undici/blob/269811b01db873f49c4d8172e95e8e82701d1f1d/lib/request.js#L118-L130

Reply:
Ok, maybe it's ok :).